core/vm/contracts.go

```diff
@@ -498,3 +505,472 @@ func intToByteArray20(n uint8) [20]byte {
 func RequiredGas(input []byte) uint64 {
 	return 0
 }
+
+// RunLockupContract is an API that ties together the Lockup Contract with the EVM.
+// The requested function is determined by input length. Contracts should ensure that their input is tightly packed
+// with abi.encodePacked.
+func RunLockupContract(evm *EVM, ownerContract common.Address, gas *uint64, input []byte) ([]byte, error) {
+	if evm.interpreter.readOnly {
+		return nil, ErrWriteProtection
+	}
+	switch len(input) {
+	case 60:
+		if err := UnwrapQi(evm, ownerContract, gas, input); err != nil {
+			return nil, err
+		} else {
+			return []byte{1}, nil
+		}
+	case 20:
+		ret, err := ClaimQiDeposit(evm, ownerContract, gas, input)
+		if err != nil {
+			return nil, err
+		} else {
+			return ret, nil
+		}
+	case 53:
+		if err := ClaimCoinbaseLockup(evm, ownerContract, evm.Context.BlockNumber.Uint64(), gas, input); err != nil {
```

Comment:
do you need to pass the block number in as evm is getting passed already?

Reply:
Fixed